warehouse/accounts/services.py

```diff
@@ -75,6 +78,19 @@ def check_password(self, userid, password):
 
         return True
 
+    def create_user(self, user):
+        if self.find_userid(user.username):
+            raise UserAlreadyExists(
+                "User with username {0} already exists!".format(user.username)
+            )
+        self.db.add(user)
+        self.db.commit()
```

Comment:
We shouldn't commit the database here, we have a single transaction per request which self.db is already a part of which will either get committed if the view is successful (if it returns a value of some kind) or rollbacked otherwise (if it raises an exception of some kind, even an HTTP exception).